### PR TITLE
hmc7044: fix clk_set_rate if channel numbers aren't continuous

### PR DESCRIPTION
**Pull request by aholtzma-am**
Previously this function treated the channel number as an index into the channel list, which worked as long as the channel list contained continuous channel numbers starting at zero. This fix looks the channel number up in the list and returns an error if not found.

### CODE REVIEW

File: drivers/frequency/hmc7044/hmc7044.c [hunk 1]
```diff
@@ -357,28 +357,36 @@ int32_t hmc7044_clk_round_rate(struct hmc7044_dev *dev, uint32_t rate,
 /**
  * Set channel rate.
  * @param dev - The device structure.
- * @param chan - Channel number.
+ * @param chan_num - Channel number.
  * @param rate - Channel rate.
  * @return SUCCESS in case of success, negative error code otherwise.
  */
-int32_t hmc7044_clk_set_rate(struct hmc7044_dev *dev, uint32_t chan,
+int32_t hmc7044_clk_set_rate(struct hmc7044_dev *dev, uint32_t chan_num,
```
Comment (cristiantpop) [inline drivers/frequency/hmc7044/hmc7044.c:364]: This seems like an unnecessary change

Comment (cristiantpop) [reply drivers/frequency/hmc7044/hmc7044.c:364]: This Commit should affect `hmc7044_clk_recalc_rate` in a similar fashion.

File: drivers/frequency/hmc7044/hmc7044.c [hunk 2]
```diff
@@ -357,28 +357,36 @@ int32_t hmc7044_clk_round_rate(struct hmc7044_dev *dev, uint32_t rate,
...
 			     uint64_t rate)
 {
 	uint32_t div;
 	int32_t ret;
+	struct hmc7044_chan_spec *chan = NULL;
 
-	if (chan >= dev->num_channels)
+	/* Find the reqested channel number */
+	for (int i=0; i < dev->num_channels; i++) {
```
Comment (cristiantpop) [inline drivers/frequency/hmc7044/hmc7044.c:372]: Declare variables at the beginning of the function

File: drivers/frequency/hmc7044/hmc7044.c [hunk 3]
```diff
@@ -637,7 +646,9 @@ static int32_t hmc7044_setup(struct hmc7044_dev *dev)
 	mdelay(1);
 	hmc7044_write(dev, HMC7044_REG_REQ_MODE_0,
 		      (dev->high_performance_mode_clock_dist_en ?
-		       HMC7044_HIGH_PERF_DISTRIB_PATH : 0));
+		       HMC7044_HIGH_PERF_DISTRIB_PATH : 0) |
```
Comment (cristiantpop) [inline drivers/frequency/hmc7044/hmc7044.c:649]: This should be in a separate commit, with description, why it is needed.

File: drivers/frequency/hmc7044/hmc7044.c [hunk 4]
```diff
@@ -804,6 +815,8 @@ int32_t hmc7044_init(struct hmc7044_dev **device,
 	dev->clkin1_vcoin_en = init_param->clkin1_vcoin_en;
 	dev->high_performance_mode_clock_dist_en =
 		init_param->high_performance_mode_clock_dist_en;
+	dev->high_performance_mode_pll_vco_en =
```
Comment (cristiantpop) [inline drivers/frequency/hmc7044/hmc7044.c:818]: This should be in a separate commit, with description, why it is needed.

Comment (dbogdan) [reply drivers/frequency/hmc7044/hmc7044.c:818]: You are not using the latest version of no-OS - this has been removed by https://github.com/analogdevicesinc/no-OS/commit/1915ff0af51ee6b65470f6a4fa4a71bab9afa884

File: drivers/frequency/hmc7044/hmc7044.c (hunk 3 above)
Comment (dbogdan) [reply drivers/frequency/hmc7044/hmc7044.c:649]: You are not using the latest version of no-OS - this has been removed by https://github.com/analogdevicesinc/no-OS/commit/1915ff0af51ee6b65470f6a4fa4a71bab9afa884

Comment (aholtzma-am) [reply drivers/frequency/hmc7044/hmc7044.c]: Apologies, this snuck in due to a convoluted merge process on our end. Reverted.

File: drivers/frequency/hmc7044/hmc7044.c (hunk 4 above)
Comment (aholtzma-am) [reply drivers/frequency/hmc7044/hmc7044.c]: Apologies, this snuck in due to a convoluted merge process on our end. Reverted.

File: drivers/frequency/hmc7044/hmc7044.c (hunk 2 above)
Comment (aholtzma-am) [reply drivers/frequency/hmc7044/hmc7044.c]: Fixed

File: drivers/frequency/hmc7044/hmc7044.c (hunk 1 above)
Comment (aholtzma-am) [reply drivers/frequency/hmc7044/hmc7044.c:374]: Done.

**Comment by adrimbarean** [comment]
Doxygen is complaining probably because parameter names on different functions have changed. Need to update those as well.

**Comment by aholtzma-am** [comment]
Parameter name in docs fixed.

**Comment by buha** [comment]
Hi @aholtzma-am ,
The change is legit, I agree there is a problem with the current implementation if the clock init param is not linear (channel num != channel index in init param).

However, please make sure cleanup the branch log:
- rebase onto master
- remove merge commits
- squash these 3 commits:
  - hmc7044: fix clk_set_rate if channel numbers aren't continuous 
  - hmc7044: revert errant changes due to bad merge
  - hmc7044: look up channel number in hmc7044_clk_recalc_rate() as well

**Comment by aholtzma-am** [comment]
Rebased and squashed.

**Review by buha — APPROVED** [review]
I've tested this on some hardware and seems there is no regression.

**Review by amiclaus — APPROVED** [review]
